### ISSUE TITLE
[SUGGESTION] Use transformation matrices instead of "fake" matrices.

### ISSUE THREAD

**Issue by TSnake41** [TurtleP/LovePotion#69]
LovePotion uses currently this structure for graphics matrices : 
```c
typedef struct
{
    double ox;
    double oy;

    double r;

    double sx;
    double sy;

    double kx;
    double ky;
} StackMatrix;
```
The problem with this is that it can be hard to stack multiples transformations (some code don't seem working (comments)) and it could be inefficient since we will have to do a pretty much computation (multiples sin+cos and some arithmetic) to transform a unique point.
With matrices, transformation have a regular cost regardless on how much do we translate (we will only have to multiply the point with a matrix which is pretty easy and optimises very well (SIMD easily)).
To stack transformations, we have to multiply a matrix with a matrix, just as simple as that.

Some useful wikipedia informations :
https://en.wikipedia.org/wiki/Transformation_matrix#Examples_in_2D_computer_graphics
https://upload.wikimedia.org/wikipedia/commons/2/2c/2D_affine_transformation_matrix.svg

About the official LÖVE way : 
They do use matrices, either 3x3 or 4x4 :
 - https://bitbucket.org/rude/love/src/825682216bf4d94220dbd9c25d0c394178627aa0/src/modules/graphics/Graphics.cpp?at=default&fileviewer=file-view-default
Their matrix implementation : https://bitbucket.org/rude/love/src/825682216bf4d94220dbd9c25d0c394178627aa0/src/common/Matrix.cpp?at=default&fileviewer=file-view-default

**Comment by NotQuiteApex** [comment]
I absolutely agree! This would require a rewrite of the PICA200 shader, and I'm not enitrely sure how that works with citro2d yet. But it would definitely take out a large chunk of code and make it much more effcient.

**Comment by TurtleP** [comment]
I know that [@fincs (creator of citro2d)](https://github.com/fincs) has said something was planned for matrix operations inside the library, but until then, we have to figure something out which works for both switch and 3ds

**Comment by TurtleP** [comment]
It's been a while since this was looked at, and there is now a pull request for citro2d for view manipulations with matrices. Furthermore, with the Switch backend moving to deko3d, this is being done. It's all in the dev/mat4 and dev/deko3d branches (although dev/deko3d has the code merged from the prior branch). I'll close and update this issue when the branch is ready for merging.

**Comment by TurtleP** [comment]
fixed with #117
